File: src/main/java/github/kasuminova/balloonserver/utils/fileobject/SimpleFileObject.java
package github.kasuminova.balloonserver.utils.fileobject;

import com.alibaba.fastjson2.annotation.JSONField;

public final class SimpleFileObject extends AbstractSimpleFileObject {
    @JSONField(ordinal = 1)
    long modified;
    @JSONField(ordinal = 2)
    String hash;
    @JSONField(ordinal = 3)
    long length;
    public SimpleFileObject(String name, long length, String hash, long modified) {
        this.name = name;
        this.length = length;
        this.hash = hash;
        this.modified = modified;
    }

    public long getLength() {
        return length;
    }

    public void setLength(long length) {
        this.length = length;
    }

    public String getHash() {
        return hash;
    }

    public void setHash(String hash) {
        this.hash = hash;
    }

    public long getModified() {
        return modified;
    }

    public void setModified(long modified) {
        this.modified = modified;
    }
}

File: src/main/kotlin/github/balloonupdate/littleserver/SimpleFileObject.kt
package github.balloonupdate.littleserver

import com.alibaba.fastjson2.JSONObject

abstract class AbstractSimpleFileObject(
    open var name: String
) {
    abstract fun toJson(): JSONObject
}

class SimpleFileObject(
    name: String,
    var length: Long,
    var hash: String,
    var modified: Long,
) : AbstractSimpleFileObject(name) {
    override fun toJson(): JSONObject {
        val obj = JSONObject()

        obj.put("name", name)
        obj.put("length", length)
        obj.put("hash", hash)
        obj.put("modified", modified)

        return obj
    }
}

class SimpleDirectoryObject(
    name: String,
    var children: ArrayList<AbstractSimpleFileObject>,
) : AbstractSimpleFileObject(name) {
    override fun toJson(): JSONObject
    {
        val obj = JSONObject()
        obj.put("name", name)

        val cs = mutableListOf<JSONObject>()
        for (child in children)
            cs += child.toJson()

        obj.put("children", cs)

        return obj
    }
}